src/jit/optcse.cpp

```diff
@@ -832,21 +847,94 @@ void Compiler::optValnumCSE_InitDataFlow()
     // We walk the set of CSE candidates and set the bit corresponsing to the CSEindex
     // in the block's bbCseGen bitset
     //
-    for (unsigned cnt = 0; cnt < optCSECandidateCount; cnt++)
+    for (unsigned inx = 0; inx < optCSECandidateCount; inx++)
     {
-        CSEdsc*      dsc      = optCSEtab[cnt];
+        CSEdsc*      dsc      = optCSEtab[inx];
         unsigned     CSEindex = dsc->csdIndex;
         treeStmtLst* lst      = dsc->csdTreeList;
         noway_assert(lst);
 
         while (lst != nullptr)
         {
-            BasicBlock* block = lst->tslBlock;
-            BitVecOps::AddElemD(cseTraits, block->bbCseGen, genCSEnum2bit(CSEindex));
+            BasicBlock* block   = lst->tslBlock;
+            unsigned    cseBit  = genCSEnum2bit(CSEindex) * 2;
+            unsigned    cseBit2 = cseBit + 1;
```

Comment:
A more descriptive name would be useful. Perhaps `cseCrossCallBit`?

Reply:
The comment above states:

```
    // two bits are allocated per CSEcandidate,
    // the first one 'cseMaskTraits' tracks normal availability
    // the second one 'cseTraits' tracks availability when calls are considered killing
```

Reply:
I am open for suggestions on what names to use here.

Reply:
I think that `cseCrossCallBit` seems reasonable, as suggested by @mikedn. 
That said, I find the comment that you reference above to be confusing. You say "two bits are allocated" and then say that the second one is `cseTraits`. But AFAICT, both bits you describe are in `cseTraits`.
I haven't fully reviewed this code yet, but it seems there are actually 3 bits per CSE candidate (2 in `cseTraits` and 1 in `cseMaskTraits`). You describe the latter as reflecting "normal availability", and I *think* the former has one bit for whether the cse is defined, and another for whether it is live across a call. If that's correct it would be good to make that clear in that comment above.

Reply:
You are right and that comment is incorrect.
These traits controls how many bits get allocated for a particular Bit Vector

There are two kinds needed now when dealing with CSEs:

`cseMaskTraits` only needs one bit per CSE candidate and it is only used by `optCSE_canSwap`
for `CSE_useMask` and `CSE_defMask` for a legality check.

I will rename the other one 'cseTraits' to 'cseLivenessTraits'. It needs two bits per CSE candidate plus one extra bit so that the liveness algorithm properly visits all children on the first visit.  

Note that I discovered this bug when I investigate couple of functions that had unexpected diffs. It previously existed, but was harder to hit because we over-allocated using BitVecTraits(EXPSET_SZ, this) on line 366, so could happen only when we have the maximum number of CSEs and an appropriate initial liveness state.

Now each CSE gets two bits, where a one bit means that the CSE is available and a 0 bit means that it is not available.   So at a definition you would see both bits changing from 0 to 1 and when visiting a call the second bit would get killed and become 0, while the other availability bit stays as 1.  I am currently calling the second bit `cseBit2`

I can call it `cseCrossCallBit` but it will go from 1 to 0 when it crosses a call.

Reply:
I renamed both bits: cseBit to cseAvailBit and cseBit2 to cseAvailCrossCallBit